src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/Breadcrumb.vue

```diff
@@ -1,25 +1,383 @@
 <template>
-  <v-breadcrumbs class="pl-0" :items="items" color="primary">
+  <v-breadcrumbs class="pl-0" :items="breadcrumbItems" color="primary">
     <template #divider>/</template>
   </v-breadcrumbs>
 </template>
 
 <script lang="ts">
-import { defineComponent, type PropType } from "vue";
+import { defineComponent } from "vue";
+import { useRoute } from "vue-router";
+import { useApplicationStore } from "@/store/application";
+import { useUserStore } from "@/store/user";
 
 export interface ItemsType {
   title: string;
   disabled: boolean;
   href: string;
 }
 
+export interface BreadcrumbItem extends ItemsType {
+  routeName?: string; // Add routeName to track which route this represents
+}
+
 export default defineComponent({
   name: "Breadcrumb",
   components: {},
-  props: {
-    items: {
-      type: Array as PropType<ItemsType[]>,
-      required: true,
+  setup() {
+    const route = useRoute();
+    const applicationStore = useApplicationStore();
+    const userStore = useUserStore();
+
+    return { route, applicationStore, userStore };
+  },
+  computed: {
+    isUnder19InApplication(): boolean {
+      return (
+        this.userStore.isUnder19 &&
+        (this.applicationStore.hasDraftApplication ||
+          this.route.name === "application-certification" ||
+          this.route.name === "application-requirements" ||
+          this.route.name === "declaration" ||
+          this.route.name === "consent-required")
+      );
+    },
+    baseItems(): ItemsType[] {
+      return [{ title: "Home", disabled: false, href: "/" }];
+    },
+    addConsentStep() {
+      return (items: ItemsType[]): ItemsType[] => {
+        if (this.isUnder19InApplication) {
+          // Find the index of the "Requirements" item
+          const requirementsIndex = items.findIndex((item) => item.title === "Requirements");
+          if (requirementsIndex !== -1) {
+            // Insert "Consent required" immediately before "Requirements"
+            const beforeRequirements = items.slice(0, requirementsIndex);
+            const afterRequirements = items.slice(requirementsIndex);
+            return [...beforeRequirements, { title: "Consent required", disabled: false, href: "/application/consent-required" }, ...afterRequirements];
+          }
+        }
+        return items;
+      };
+    },
+
+    // Define the ranking for application pages
+    applicationPageRankings(): Record<string, number> {
+      return {
+        "application-certification": 1,
+        "application-transfer": 1, // Same rank as certification since they're alternatives
+        "consent-required": 2,
+        "application-requirements": 3,
+        declaration: 4,
```

Comment:
is this a github bug? not sure why it's showing as having no quotes in the preview. 

<img width="323" height="60" alt="image" src="https://github.com/user-attachments/assets/f3330068-312f-4215-bb62-5d4c8f4b185b" />

Reply:
No, I believe this is expected and prettier formatting as`declaration` without the hyphen is a valid json identifier. Whereas, for `"application-requirements"` quotes are required to make with hyphen key a valid json identifier.